Feature gradlerio

inyongie opened the pull request:
pullll requesttttt

Commit:
initial commit, work in progress

inyongie committed:
GradleRIO setup process ongoing

inyongie committed:
adding GradleRIO integrated build.gradle.

admin4571 commented on build.gradle:
```diff
@@ -1,37 +1,63 @@
-//------------------------------------------- PLUGIN DEFINITION -----------------------------------------------//
-
```
Can you add back all comments in this file? I want to use this as a template to explain it to the students.

inyongie replied:
will do

admin4571 commented on build.gradle:
```diff
@@ -1,37 +1,63 @@
+buildscript {
+  repositories {
+    //mavenLocal()
+	mavenCentral()
```
Do we not need maven local anymore?

inyongie replied:
I would say it's more for us than them, as long as there's internet it's not an issue

arjunrao87 replied:
I would just leave it there because it does the waterfall logic in any case.

admin4571 commented on build.gradle:
```diff
@@ -1,37 +1,63 @@
+//    id 'idea'
+//    id 'eclipse'
+//    id 'jacoco'
+//}
```
Lets delete these commented lines

inyongie replied:
will do

admin4571 commented on build.gradle:
```diff
@@ -1,37 +1,63 @@
-	compile files('libs/WPILib.jar', 'libs/NetworkTables.jar')
-	compile 'org.apache.commons:commons-collections4:4.1'
-    testCompile group: 'junit', name: 'junit', version: '4.+'
+task genJavadoc(type: Jar, dependsOn: javadoc) {
```
I dont quite understand how we will get the dependencies that we are removing here. If we get rid of the compile files lines, I am expecting the Travis build to break. Isnt that right?

inyongie replied:
The "GradleRIO" repository added after mavenCentral seems to pull the WPI libs

admin4571 replied:
And how do the other dependencies get pulled in? like say commons-collections?

inyongie replied:
classpath group:<groupName>, name: <name>, version: <version>

If you make
testCompile group: 'junit', name: 'junit', version: '4.+'
into
classpath group: 'junit', name: 'junit', version: '4.+'

it worked.

admin4571 commented on build.gradle:
```diff
@@ -41,32 +67,25 @@ sourceSets {
-  attributes 'Implementation-Title': 'Crate Robot'
-}
-
-task fatjar( type: Jar ) {
```
why has the fatjar task been deleted? I think we should keep it in even with the gradle rio deploy artifact

inyongie replied:
it's defined as the "jar" task in the GradleRIO's gradle.build. It's there, looks like the same underlying logic.

admin4571 replied:
Im fine with that as long as we add baseName = project.name from the fatjar task into the jar task. That way when the jar gets built, we are differentiating it between a prod and dev version.

admin4571 commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.2.1-bin.zip
```
Why did we have to downgrade the gradle version ?

inyongie replied:
just following GradleRIO's version. If latest works, should be fine to use the newer.

admin4571 replied:
Agreed

admin4571 commented on gradlew:
```diff
@@ -6,30 +6,12 @@
-SAVED="`pwd`"
-cd "`dirname \"$PRG\"`/" >/dev/null
-APP_HOME="`pwd -P`"
-cd "$SAVED" >/dev/null
```
Have we just moved this section down in the file?

inyongie replied:
I copy-pasted 2 different versions of gradlew, both of them as-given and no mods by me.

admin4571 commented on gradlew:
```diff
@@ -89,7 +90,7 @@ location of your Java installation."
 
 # Increase the maximum file descriptors if we can.
-if [ "$cygwin" = "false" -a "$darwin" = "false" -a "$nonstop" = "false" ] ; then
+if [ "$cygwin" = "false" -a "$darwin" = "false" ] ; then
```
Not exactly sure what the implication of this change is

inyongie replied:
I copy-pasted 2 different versions of gradlew, both of them as-given and no mods by me.

admin4571 replied:
So this came as part of the initial defaults. I guess unless we have no reason to add it back, we can leave it this way. Ack.

admin4571 commented on gradlew:
```diff
@@ -113,7 +114,6 @@ fi
     APP_HOME=`cygpath --path --mixed "$APP_HOME"`
     CLASSPATH=`cygpath --path --mixed "$CLASSPATH"`
-    JAVACMD=`cygpath --unix "$JAVACMD"`
 
```
Or this one

inyongie replied:
I copy-pasted 2 different versions of gradlew, both of them as-given and no mods by me.

admin4571 replied:
ok. nobody cares about cygwin anyway

admin4571 commented on gradlew.bat:
```diff
@@ -1,90 +1,90 @@
+
+:init
+@rem Get command-line arguments, handling Windowz variants
+
```
Lol

inyongie replied:
I copy-pasted 2 different versions of gradlew, both of them as-given and no mods by me.

admin4571 replied:
the lol was for "Windows -> Windowz"

Commit:
changing build.gradle according to pull request

Commit:
header comment changes

inyongie commented:
did some commits, check them out

admin4571 commented on build.gradle:
```diff
@@ -1,37 +1,63 @@
+//    compile 'org.apache.commons:commons-collections4:4.1'
+//    testCompile group: 'junit', name: 'junit', version: '4.+'
+  }
 }
```
Same as below. How do the other dependencies get pulled in? like say commons-collections? I dont wanna have this commented piece here because i want to make sure we know how to import external dependencies. This is a good time to find out.

inyongie replied:
classpath group:, name: , version:

If you make
testCompile group: 'junit', name: 'junit', version: '4.+'
into
classpath group: 'junit', name: 'junit', version: '4.+'

it worked.


I'll take the comment out.

admin4571 reviewed (changes requested):
Minor changes. Should be ready for a merge after this.

inyongie committed:
for PR comments.

inyongie committed:
Toast is loaded. Removing the libs/ jars, as Toast/GradleRIO dependency seems to contain these.

inyongie committed:
Adding build info to README.md

inyongie commented:
continuous-integration/travis-ci/pr is failing with the logs saying that it failed tests, but there are no tests written in the TestClass yet?

arjunrao87 commented:
The travis build is failing because of something I was playing around with CodeCov. I am going to revert the changes.

inyongie committed:
Changing base class to extend IterativeModule, which is Toast's version of IterativeRobot.

inyongie commented:
Is this CodeCov issue still around?
